ide/html.editor/src/org/netbeans/modules/html/editor/completion/HtmlCompletionProvider.java

```diff
@@ -386,7 +386,16 @@ public void run() {
                             } else {
                                 ts.move(dotPos - 1);
                                 if (ts.moveNext() || ts.movePrevious()) {
-                                    if (ts.token().id() == HTMLTokenId.TAG_CLOSE_SYMBOL && !CharSequenceUtilities.equals("/>", ts.token().text())) {
+                                    if (!CharSequenceUtilities.equals("/>", ts.token().text()) && null != LexerUtils.followsToken(ts, HTMLTokenId.TAG_OPEN, true, false,
+                                            HTMLTokenId.ARGUMENT,
+                                            HTMLTokenId.VALUE,
+                                            HTMLTokenId.VALUE_CSS,
+                                            HTMLTokenId.VALUE_JAVASCRIPT,
+                                            HTMLTokenId.OPERATOR,
+                                            HTMLTokenId.WS,
+                                            HTMLTokenId.EL_CLOSE_DELIMITER,
+                                            HTMLTokenId.EL_CONTENT,
+                                            HTMLTokenId.EL_OPEN_DELIMITER)) {
```

Comment:
In general I like this - the current behavior annoyed the hell out of me. The second check looks to complex - If I understand correctly, you want to check if you are behind an open tag. I would invert this and check if it is not a closing tag. 

```suggestion
                                    if (!CharSequenceUtilities.equals("/>", ts.token().text()) && null == LexerUtils.followsToken(ts, HTMLTokenId.TAG_CLOSE, true, false,
                                            HTMLTokenId.WS,
                                            HTMLTokenId.TAG_CLOSE_SYMBOL)) {
```

The idea here is, that the definition for a closing tag is pretty simple: https://www.w3.org/html/wg/spec/syntax.html#end-tags

Reply:
Shouldn't it be null !=... I thought it gives you the closing tag back, if the method finds it? But in general you are right, less checks etc. seems more performant and cleaner.

Reply:
The suggested check returns true if we are _not_ directly behind a closing tag. `null !=` would yield true after a closing tag, which is the opposite of what we want.

Reply:
True, got it. Thx for the hint. I fixed and tested it. Worked as before.